cpu/stm32/periph/i2c_2.c

```diff
@@ -292,30 +326,32 @@ int i2c_write_bytes(i2c_t dev, uint16_t address, const void *data,
 
     /* Send out data bytes */
     for (size_t i = 0; i < length; i++) {
-        DEBUG("[i2c] write_bytes: Waiting for TX reg to be free\n");
+        DEBUG_PUTS("[i2c] i2c_write_bytes(): Waiting for TX reg to be free");
         ret = _is_sr1_mask_set(i2c, I2C_SR1_TXE, flags);
         if (ret < 0) {
+            DEBUG_PUTS("[i2c] i2c_write_bytes(): Waiting for I2C_SR1_TXE failed");
             return ret;
         }
-        DEBUG("[i2c] write_bytes: TX is free so send byte\n");
+        DEBUG_PUTS("[i2c] i2c_write_bytes(): TX is free so send byte");
         i2c->DR = ((uint8_t*)data)[i];
     }
     /* Wait for tx reg to be empty so other calls will no interfere */
     ret = _is_sr1_mask_set(i2c, I2C_SR1_TXE, flags);
     if (ret < 0) {
+        DEBUG_PUTS("[i2c] i2c_write_bytes(): Waiting for I2C_SR1_TXE failed");
         return ret;
     }
     if (flags & I2C_NOSTOP) {
         return 0;
     }
     else {
         /* End transmission */
-        DEBUG("[i2c] write_bytes: Ending transmission\n");
+        DEBUG_PUTS("[i2c] i2c_write_bytes(): Ending transmission");
```

Comment:
why not stay with DEBUG

afaik you optimized the behavior of printf -> DEBUG on AVR  (I know this isn't AVR).

a short visit to godbolt revealed printf is optimized to puts anyway  ( even with gcc -O0)  (probably not for AVR with your flashprint optimization )

Reply:
```
#ifdef DEVELHELP
#include "cpu_conf.h"
#define DEBUG_PRINT(...) \
    do { \
        if ((thread_get_active() == NULL) || \
            (thread_get_active()->stack_size >= \
             THREAD_EXTRA_STACKSIZE_PRINTF)) { \
            printf(__VA_ARGS__); \
        } \
        else { \
            puts("Cannot debug, stack too small. Consider using DEBUG_PUTS()."); \
        } \
    } while (0)
#else
#define DEBUG_PRINT(...) printf(__VA_ARGS__)
#endif
```
wat

Reply:
fixed that in #20166